### PR TITLE
added default case for normal to mesh projection manifold

### PR DESCRIPTION
**Pull request by nicola-giuliani**
I have noticed that NormalToMeshProjectionManifold lacks a default case, it only implements case 2, 4, 8. I have added a simple default scenario to take care of any strange projection case. I have also added a new test to check the implementation.

### CODE REVIEW

File: source/opencascade/manifold_lib.cc [hunk 1]
```diff
@@ -302,7 +302,43 @@ namespace OpenCASCADE
...
+                        Tensor<1, 3> n1(n_coords);
+                        if (n1.norm() > tolerance)
+                          {
+                            n1 = n1 / n1.norm();
+                            if (average_normal.norm() < tolerance)
+                              average_normal = n1;
+                            else
+                              {
+                                auto dot_prod = n1 * average_normal;
+                                // We check the
```
Comment (luca-heltai) [inline source/opencascade/manifold_lib.cc]:
```suggestion
                                // We check that the direction of the normal vector w.r.t the 
                                // current average, and make sure we flip it if it is opposite 
```

File: source/opencascade/manifold_lib.cc [hunk 2]
```diff
@@ -302,7 +302,43 @@ namespace OpenCASCADE
...
+                                  average_normal += n1;
+                                else
+                                  average_normal -= n1;
+                              }
+                          }
+                      }
+            Assert(
+              average_normal.norm() > tolerance,
+              ExcMessage(
+                "Failed to refine cell: the normal estimated via the surrounding points turns out to be a null vector, making the projection direction undetermined."));
```
Comment (luca-heltai) [inline source/opencascade/manifold_lib.cc]:
```suggestion
                "Failed to compute a normal: the normal estimated via the surrounding points turns out to be a null vector, making the projection direction undetermined."));
```

File: tests/opencascade/normal_to_mesh_projection_04.cc [hunk 3]
```diff
@@ -0,0 +1,101 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2014 - 2019 by the deal.II authors
```
Comment (luca-heltai) [inline tests/opencascade/normal_to_mesh_projection_04.cc]:
```suggestion
// Copyright (C) 2019 by the deal.II authors
```

File: tests/opencascade/normal_to_mesh_projection_04.cc [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Create a BSpline surface, and test axis normal to mesh projection using 5
```
Comment (luca-heltai) [inline tests/opencascade/normal_to_mesh_projection_04.cc]:
```suggestion
// Create a BSpline surface, and test NormalToMeshProjectionManifold  projection using 5
```

**Comment by luca-heltai** [comment]
Just minor suggestions.

**Comment by luca-heltai** [comment]
Can you squash the commits?

**Comment by nicola-giuliani** [comment]
just did

**Pull request merged**